lib/state/ui/reducer.ts

```diff
@@ -192,15 +192,21 @@ const openedRevision: A.Reducer<[T.EntityId, number] | null> = (
   }
 };
 
-const openedTag: A.Reducer<T.TagHash | null> = (state = null, action) => {
+const showCollection: A.Reducer<T.Collection> = (
+  state = { type: 'all' },
+  action
+) => {
   switch (action.type) {
     case 'SELECT_TRASH':
+      return { type: 'trash' };
     case 'SHOW_ALL_NOTES':
-      return null;
+      return { type: 'all' };
     case 'OPEN_TAG':
-      return tagHashOf(action.tagName);
+      return { type: 'tag', tagHash: tagHashOf(action.tagName) };
     case 'TRASH_TAG':
-      return tagHashOf(action.tagName) === state ? null : state;
+      return tagHashOf(action.tagName) === state.tagHash
+        ? { type: 'all' }
+        : state;
```

Comment:
love how we can make this clear update based on the deletion of a tag without having to worry about updating a few different elements in different places.

Reply:
It definitely is more descriptive :)